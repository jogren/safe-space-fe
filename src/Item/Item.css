.Item {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 100%;
}

#check-or-uncheck {
  height: 20px;
  width: 25%;
  margin-left: 15%;
}

#item-name {
  width: 50%;
}

#new-item {
  font-size: 20px;
  padding-left: 5px;
  width: 50%;
  height: 50px;
  border-radius: 5px;
}

#item-name:hover {
  background-color: white;
  color: black;
}

#checkbox-item-container {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}
/* #dietary-item,
#medications-item,
#allergies-item,
#other-item {
  margin-left: 9%;
} */
