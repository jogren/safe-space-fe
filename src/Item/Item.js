import React, { useState } from "react";
import "./Item.css";

const Item = ({ item, items, setItems, neededItems, setNeededItems }) => {
  const [newItem, setNewItem] = useState("");

  const handleChange = e => setNewItem(e.target.value);

  const handleSubmit = (e, newItem) => {
    e.preventDefault();
    items.splice(items.length - 1, 0, newItem);
    setItems([...items]);
    setNeededItems([...neededItems, newItem]);
  };

  const handleCheck = e => {
    if (!neededItems.includes(item)) {
      setNeededItems([...neededItems, item]);
    } else {
      let filteredItems = neededItems.filter(itemName => itemName !== item);
      setNeededItems(filteredItems);
    }
  };

  return (
    <article className="Item">
      {item === "add item" ? (
        <div id="add-item-container">
          <input
            id="new-item"
            type="text"
            placeholder="Add Item/Medication"
            name="name"
            value={newItem}
            onChange={handleChange}
            autoComplete="off"
          />
          <button disabled={!newItem} id="submit-new-item" type="submit" onClick={e => handleSubmit(e, newItem)}>
            <img
              id="plus-img"
              alt="plus symbol"
              src={require("../assets/plus-sign.svg")}
            />
          </button>
        </div>
      ) : (
        <div id="checkbox-item-container">
          <img
            id="check-or-uncheck"
            alt=""
            src={
              !neededItems.includes(item)
                ? require("../assets/unchecked.svg")
                : require("../assets/checked.svg")
            }
            onClick={handleCheck}
          />
          <h3 id="item-name" onClick={handleCheck}>
            {item}
          </h3>
        </div>
      )}
    </article>
  );
};

export default Item;
