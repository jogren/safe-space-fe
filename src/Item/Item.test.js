import React from "react";
import { shallow } from "enzyme";
import Item from "./Item";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Item", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Item
        item=""
        items={[
          "Diapers",
          "Baby Wipes",
          "Breastfeeding Supplies",
          "Infant Formula",
          "Feminine Products",
          "Phone Charger (iphone)",
          "Phone Charger (android)",
          "add item"
          ]}
        setItems={jest.fn()}
        neededItems={[]}
        setNeededItems={jest.fn()}
      />
    );
  });

  it("should render without crashing", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match the snapshot with three empty name, age, and phone inputs passed through", () => {
    expect(wrapper).toMatchSnapshot();
  });
});