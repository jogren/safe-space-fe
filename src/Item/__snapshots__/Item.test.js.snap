// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match the snapshot with three empty name, age, and phone inputs passed through 1`] = `
<article
  className="Item"
>
  <div
    id="checkbox-item-container"
  >
    <img
      alt=""
      id="check-or-uncheck"
      onClick={[Function]}
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "unchecked.svg",
        }
      }
    />
    <h3
      id="item-name"
      onClick={[Function]}
    />
  </div>
</article>
`;
